core/server/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -269,7 +270,14 @@ public void accessBlock(long sessionId, long blockId) throws BlockDoesNotExistEx
   public void commitBlock(long sessionId, long blockId)
       throws BlockAlreadyExistsException, BlockDoesNotExistException, InvalidWorkerStateException,
       IOException, WorkerOutOfSpaceException {
-    mBlockStore.commitBlock(sessionId, blockId);
+    // NOTE: this may be invoked multiple times due to retry on client side.
+    // TODO(binfan): find a better way to handle retry logic
+    try {
+      mBlockStore.commitBlock(sessionId, blockId);
+    } catch (BlockAlreadyExistsException e) {
+      LOG.debug("Block {} has been in block store, this could be a retry due to master-side RPC "
+          + "failure, therefore ignore the exception", blockId, e);
+    }
 
     // TODO(calvin): Reconsider how to do this without heavy locking.
     // Block successfully committed, update master with new block metadata
```

Comment:
Is updating master a second time with the same block metadata going to cause problems?